### PR TITLE
Rewrites/coding convention

### PR DESCRIPTION
**Pull request by Quix0r**
I have again applied more coding convention stuff, e.g. added missing curly braces, spaces and usage of `dbm::is_result()`. Also I have unified most db upgrades to a more common way:

````
function update_xxxx() {
    $r = q("ALTER TABLE `foo` DO SOMETHING");

    if ($r) {
        return UPDATE_SUCCESS;
    }
    return UPDATE_FAILED;
}
````
There are a few exceptions, plus a lot older updates that needs being fixed/considered.

### CODE REVIEW

File: mod/wall_upload.php [hunk 1]
```diff
@@ -228,14 +228,14 @@ function wall_upload_post(App $a, $desktopmode = true) {
 	if($width > 640 || $height > 640) {
 		$ph->scaleImage(640);
 		$r = $ph->store($page_owner_uid, $visitor, $hash, $filename, t('Wall Photos'), 1, 0, $defperm);
-		if($r)
+		if ($r)
```
Comment (Hypolite) [inline mod/wall_upload.php]: Standards: Please add braces to this conditional statement.

File: mod/wall_upload.php [hunk 2]
```diff
@@ -228,14 +228,14 @@ function wall_upload_post(App $a, $desktopmode = true) {
...
 			$smallest = 1;
 	}
 
-	if($width > 320 || $height > 320) {
+	if ($width > 320 || $height > 320) {
 		$ph->scaleImage(320);
 		$r = $ph->store($page_owner_uid, $visitor, $hash, $filename, t('Wall Photos'), 2, 0, $defperm);
-		if($r AND ($smallest == 0))
+		if ($r AND ($smallest == 0))
```
Comment (Hypolite) [inline mod/wall_upload.php]: Standards: Please add braces to this conditional statement.

File: update.php [hunk 3]
```diff
@@ -308,7 +308,7 @@ function update_1030() {
 function update_1031() {
 	// Repair any bad links that slipped into the item table
 	$r = q("SELECT `id`, `object` FROM `item` WHERE `object` != '' ");
-	if($r && dbm::is_result($r)) {
+	if (dbm::is_result($r)) {
 		foreach ($r as $rr) {
 			if(strstr($rr['object'],'type=&quot;http')) {
```
Comment (Hypolite) [inline update.php]: Standards: Please add a space before the ```if```.

File: update.php [hunk 4]
```diff
@@ -567,6 +567,7 @@ function update_1069() {
 	q("ALTER TABLE `fcontact` ADD `request` CHAR( 255 ) NOT NULL AFTER `photo` ");
 }
 
+/// @TODO Still meeded?
```
Comment (Hypolite) [inline update.php]: I'm not sure the TODO tag is the best option to discuss code changes. I would expect a TODO to outline a future change, not to ask a question.

File: update.php [hunk 5]
```diff
@@ -886,6 +895,7 @@ function update_1102() {
 
 
 function update_1103() {
+/// @TODO Comented out:
```
Comment (Hypolite) [inline update.php]: Typo, should be "Commented". What's the exact purpose of this TODO tag?

File: update.php [hunk 6]
```diff
@@ -1031,9 +1041,11 @@ function update_1120() {
 
 	$r = q("describe item");
 	if (dbm::is_result($r)) {
-		foreach($r as $rr)
-			if($rr['Field'] == 'spam')
+		foreach($r as $rr) {
```
Comment (Hypolite) [inline update.php]: Standards: Please add a space after the ```foreach```.

File: update.php (hunk 3 above)
Comment (Quix0r) [inline update.php]: ??? Is already added. Or do you mean an empty line?

File: update.php (hunk 4 above)
Comment (Quix0r) [inline update.php]: Removed

File: update.php (hunk 5 above)
Comment (Quix0r) [inline update.php]: Fixed tpzo ... ;-) but the original committer is @friendica some how? commit id b795ce4f

File: update.php (hunk 3 above)
Comment (annando) [inline update.php]: See line 313. I guess he meant that.

Comment (Hypolite) [inline update.php]: @annando is correct. 😊

Comment (Quix0r) [inline update.php]: Okay found + fixed it.

File: mod/crepair.php [hunk 7]
```diff
@@ -78,18 +82,18 @@ function crepair_post(App $a) {
...
-	if($photo) {
+	if ($photo) {
 		logger('mod-crepair: updating photo from ' . $photo);
 		require_once("include/Photo.php");
 
 		update_contact_avatar($photo,local_user(),$contact['id']);
 	}
 
-	if($r)
+	if ($r) {
```
Comment (annando) [inline mod/crepair.php]: What about "dbm::is_result"?

File: mod/message.php [hunk 8]
```diff
@@ -262,8 +262,9 @@ function message_content(App $a) {
 				//	);
 				//}
 
-				if($r)
+				if ($r) {
```
Comment (annando) [inline mod/message.php]: Isn't "$r" the result of a query?

File: mod/profiles.php [hunk 9]
```diff
@@ -34,8 +34,9 @@ function profiles_init(App $a) {
 			intval($a->argv[2]),
 			intval(local_user())
 		);
-		if($r)
+		if ($r) {
```
Comment (annando) [inline mod/profiles.php]: See my comment from above.

File: mod/profiles.php [hunk 10]
```diff
@@ -473,11 +473,12 @@ function profiles_post(App $a) {
 			intval(local_user())
 		);
 
-		if($r)
+		if ($r) {
```
Comment (annando) [inline mod/profiles.php]: You missed several of them ;.)

File: mod/settings.php [hunk 11]
```diff
@@ -381,24 +381,27 @@ function settings_post(App $a) {
 			$err = true;
         }
```
Comment (annando) [inline mod/settings.php]: This line is wrong intended as well.

File: mod/settings.php [hunk 12]
```diff
@@ -381,24 +381,27 @@ function settings_post(App $a) {
...
-        if( $oldpass != $r[0]['password'] ) {
-            notice( t('Wrong password.') . EOL);
-            $err = true;
-        }
+		//  check if the old password was supplied correctly before
+		//  changing it to the new value
+		$r = q("SELECT `password` FROM `user`WHERE `uid` = %d LIMIT 1", intval(local_user()));
+		if (!dbm::is_result($r)) {
+			killme();
+		} elseif ( $oldpass != $r[0]['password'] ) {
```
Comment (annando) [inline mod/settings.php]: Spaces inside the brackets aren't nice as well, I think.

File: mod/settings.php [hunk 13]
```diff
@@ -381,24 +381,27 @@ function settings_post(App $a) {
...
+		}
 
 		if(! $err) {
 			$password = hash('whirlpool',$newpass);
 			$r = q("UPDATE `user` SET `password` = '%s' WHERE `uid` = %d",
 				dbesc($password),
 				intval(local_user())
 			);
-			if($r)
+			if ($r) {
```
Comment (annando) [inline mod/settings.php]: The results of update queries should be checked via "dbm:is_result" as well, shouldn't it?

File: mod/settings.php [hunk 14]
```diff
@@ -599,8 +602,9 @@ function settings_post(App $a) {
 			dbesc($language),
 			intval(local_user())
 	);
-	if($r)
+	if ($r) {
```
Comment (annando) [inline mod/settings.php]: See my comment from above.

File: util/po2php.php [hunk 15]
```diff
@@ -12,7 +12,7 @@ function po2php_run(&$argv, &$argc) {
 	$pofile = $argv[1];
 	$outfile = dirname($pofile)."/strings.php";
 
-	if(strstr($outfile,'util'))
+	if (strstr($outfile,'util'))
```
Comment (annando) [inline util/po2php.php]: There are missing brackets behind the "if" and at the "else".

File: mod/crepair.php (hunk 7 above)
Comment (Quix0r) [inline mod/crepair.php]: That came from an UPDATE statement and I have not clearly checked it that I can really use `dbm::is_result()` here.

File: mod/message.php (hunk 8 above)
Comment (Quix0r) [inline mod/message.php]: Similar with here, DELETE is the statement (and a typical candiate is). I'm not sure if `dbm::is_result()` can be really used here.

**Comment by Quix0r** [comment]
I noticed that some $ went missing, but maybe only a local change. I fixed them anyway ...

And I want to (later, not here) add much more type-hints to method/function signatures, I noticed many can be set (avoids programming mistakes).

File: include/ostatus.php [hunk 16]
```diff
@@ -238,12 +245,14 @@ public static function import($xml,$importer,&$contact, &$hub) {
 
 		$gub = "";
 		$hub_attributes = $xpath->query("/atom:feed/atom:link[@rel='hub']")->item(0)->attributes;
-		if (is_object($hub_attributes))
-			foreach($hub_attributes AS $hub_attribute)
+		if (is_object($hub_attributes)) {
+			foreach($hub_attributes AS $hub_attribute) {
```
Comment (Quix0r) [inline include/ostatus.php]: I can generally change it in the whole file. Then I don't need to make another PR for the same file and rewrite step.

File: mod/contacts.php [hunk 17]
```diff
@@ -212,17 +213,20 @@ function contacts_post(App $a) {
 		intval($contact_id),
 		intval(local_user())
 	);
-	if($r)
+	if ($r) {
```
Comment (Quix0r) [inline mod/contacts.php]: Same with others, came from an UPDATE statement.

File: mod/settings.php (hunk 13 above)
Comment (Quix0r) [inline mod/settings.php]: I have not decided it yet as UPDATE/DELETE/INSERT don't return a result (=rows).

**Comment by Quix0r** [comment]
More changes to "base" directory (root):

- [ ] https://github.com/Quix0r/friendica/tree/rewrites/coding_convention_base

**Comment by Quix0r** [comment]
Have not yet decided what to do with UPDATE/INSERT/DELETE, your choices? Else has been addressed.

**Comment by annando** [comment]
Please wait with the other pull requests for the time after the release. We are now in feature freeze. And since these rewrites can introduce new bugs we should now concentrate on accepting the six pending pull requests.

Additionally we have to concentrate on fixing the open bugs.

I'm fine with not changing the update/insert/delete stuff by now if that speeds up the process.We can do that somewhere in the future as well.

**Comment by Quix0r** [comment]
@annando okay, will wait with them. The only thing is that my develop branch is at _rewrites/coding-convention-include-foreach_. So there will be more merge commits.

**Comment by Quix0r** [comment]
Argh, rebasing + push delete seems to close this. :-(

**Comment by tobiasd** [comment]
Was this intended for 3.5.1 or 3.5.2 @Quix0r @annando? If for 3.5.1, does #3112 need something or can it be merged?

**Comment by annando** [comment]
I wouldn't like to have huge changes for the next version. So it would be best to have it accepted before. But I would like to go through the changes one more time. I did it for this pull request. But since it is now another one, I want to do it new.

**Comment by Quix0r** [comment]
Sorry for the mess. :-( I forgot it that when I delete a remote branch, the linked PR is being auto-closed as well. :-(

**Comment by tobiasd** [comment]
3112 is closed for you @Quix0r?

**Comment by Quix0r** [comment]
@tobiasd no, still open. :-)

**Comment by tobiasd** [comment]
Then I totally don't understand your comment from Jan 31

**Comment by Quix0r** [comment]
*this* linked PR has been auto-closed by github (because the underlaying branch is gone).

**Pull request closed without merging**